File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/workout/model/SetDetails.kt
package pt.isel.leic.ptgest.domain.workout.model

import pt.isel.leic.ptgest.domain.workout.SetType

data class SetDetails(
    val id: Int,
    val name: String,
    val notes: String?,
    val type: SetType,
    val setExerciseDetails: List<SetExerciseDetails>
) {
    constructor(set: Set, setExerciseDetails: List<SetExerciseDetails>) :
        this(set.id, set.name, set.notes, set.type, setExerciseDetails)
}

File: code/js/src/views/user/TrainerViews/models/sets/SetDetails.ts
import SetExerciseDetails from "./SetExerciseDetails.ts"

class SetDetails {
    id: number
    name: string
    notes: string | null
    type: string
    orderId: number
    setExerciseDetails: SetExerciseDetails[]

    constructor(id: number, name: string, notes: string, type: string, orderId: number,setExerciseDetails: SetExerciseDetails[]) {
        this.id = id
        this.name = name
        this.notes = notes
        this.type = type
        this.orderId = orderId
        this.setExerciseDetails = setExerciseDetails
    }
}

export default SetDetails
